src/expertsystem/reaction/topology.py

```diff
@@ -425,10 +425,27 @@ def build(
 
         logging.info("finished building topology graphs...")
         # strip the current open end edges list from the result graph tuples
-        topologies: Tuple[Topology, ...] = tuple()
+        topologies = list()
         for graph_tuple in graph_tuple_list:
-            topologies += (graph_tuple[0].freeze(),)
-        return topologies
+            topology = graph_tuple[0].freeze()
+            new_edges = dict()
+            # Relabel so that initial edge IDs are [-1, -2, ...]
+            for new_edge_id, edge_id in zip(
+                range(-1, -len(topology.incoming_edge_ids) - 1, -1),
+                topology.incoming_edge_ids,
+            ):
+                new_edges[new_edge_id] = topology.edges[edge_id]
+            # Relabel so that
+            # outgoing edge IDs are [0, 1, 2, ..., n]
+            # intermediate edge IDs are [n+1, n+2, ...]
+            for new_edge_id, edge_id in enumerate(
+                tuple(topology.outgoing_edge_ids)
+                + tuple(topology.intermediate_edge_ids)
+            ):
+                new_edges[new_edge_id] = topology.edges[edge_id]
+            topology = attr.evolve(topology, edges=new_edges)
+            topologies.append(topology)
```

Comment:
Maybe put this into a separate function

Reply:
I tried to design a `relabel_edge` method c77374f, but that resulted in bugs

Reply:
a61bff0 offers a more restricted relabeling

Reply:
Yeah thats ok. I was just referring to a closure style function, to be able to refactor things a bit easier later on.